### DISCUSSION TITLE
url path is not encoded when deep quering

### DISCUSSION THREAD

**Discussion by gubarez** [payloadcms/payload#1972] [Roadmap]
# Bug Report

The url path is not encoded when deep quering. Somewhere there should be an encodeURI when building the url path.

## Steps to Reproduce

`
const doc = await payload.findByID({
        collection,
        id: doc.id,
        depth: 1,
        showHiddenFields: true
})

doc.image.url == "http://localhost:3000/media/test image.jpg" //true

`

## Other Details

Version 1.5.9

**Comment by jmikrut** [comment]
Hey @gubarez — we are not currently encoding URLs to media but we certainly could. We will slate this into our workload and get them encoded for you. That is a good call. 

Is this preventing you from having something functional? AFAIK you could URL-encode in your frontend where necessary but I actually have never had to do that manually so I'm curious as to what you ran into.

I will convert this to a feature request and add it to our roadmap. Should be able to get to it fairly quickly!

**Comment by JarrodMFlesch** [comment]
If this comes up again we can tackle it then.
